packages/p2p-v2/src/hooks/useAdvertiserStats.ts

```diff
@@ -0,0 +1,57 @@
+import { useMemo } from 'react';
+import { useAdvertiserInfo } from '@deriv/api';
+
+/**
+ * Hook to calculate an advertiser's stats based on their information.
+ *
+ * @param advertiserId - ID of the advertiser stats to reveal. If not provided, by default it will return the user's own stats.
+ */
+const useAdvertiserStats = (advertiserId?: string) => {
+    const { data, isSuccess } = useAdvertiserInfo(advertiserId);
+
+    const transformedData = useMemo(() => {
+        if (!isSuccess) return;
+
+        return {
+            /** The average buy time in minutes */
+            averagePayTime: data?.buy_time_avg && data.buy_time_avg > 60 ? Math.round(data.buy_time_avg / 60) : 1,
```

Comment:
wont this work `averagePayTime: data?.buy_time_avg > 60 ? Math.round(data.buy_time_avg / 60) : 1` ?

Reply:
Noo it was giving a typescript error, I had to explicitly guard check it

Reply:
![Uploading Screenshot 2024-01-08 at 5.01.40 PM.png…]()